### PR TITLE
[ML-27370] Add batch score inference using feature store

### PR DESCRIPTION
**Pull request by zhe-db**
Add support to use `score_batch` for batch inference if feature store enabled.

Tests: manual test with Azure-MLOps-CUJ

### CODE REVIEW

File: {{cookiecutter.project_name}}/steps/predict.py [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+    """
+    Apply the model at the specified URI for batch inference on the table with name input_table_name,
+    writing results to the table with name output_table_name
+    """
+    table = spark_session.table(input_table_name)
 
+    prediction_df = fs_client.score_batch(
+        model_uri,
+        table
+    )
```
Comment (mingyu89) [inline {{cookiecutter.project_name}}/steps/predict.py:17]: With or without feature store, the only difference is the predict step.
You can keep BatchInference file the same and only modify inside the predict_batch function.

```
...
table = spark_session.table(input_table_name)

{% if cookiecutter.include_feature_store == "yes" %}
    from databricks.feature_store import FeatureStoreClient

    fs_client = FeatureStoreClient()
    prediction_df = fs_client.score_batch(
        model_uri,
        table
    )
    prediction = prediction_df["prediction"]
{% else %}
    predict = mlflow.pyfunc.spark_udf(
        spark_session, model_uri, result_type="string", env_manager="conda"
    )
    prediction = predict(struct(*table.columns))
{% endif %}

    output_df = (
        table.withColumn("prediction", prediction)
    ...
```

Comment (zhe-db) [reply {{cookiecutter.project_name}}/steps/predict.py:21]: sounds good

Comment (vladimirk-db) [reply {{cookiecutter.project_name}}/steps/predict.py:21]: Aren't all the steps cleaned up if feature store is enabled? Can you please double check?

Comment (mingyu89) [reply {{cookiecutter.project_name}}/steps/predict.py:21]: @vladimirk-db Great question! We are leaving the steps in the repo when feature store is enabled. Partial reason is that the folder also contains functions not related to recipes, for example, deploy.py, utils.py.
After the refactor-layout PR, the steps folder will only contain recipes functions and we can remove it.

Comment (vladimirk-db) [reply {{cookiecutter.project_name}}/steps/predict.py:21]: Sounds good, just making sure we don't prune this file by accident :-)

**Review by vladimirk-db — APPROVED** [review]

File: {{cookiecutter.project_name}}/steps/predict.py [hunk 2]
```diff
@@ -10,15 +10,29 @@ def predict_batch(
...
         table.withColumn("prediction", predict(struct(*table.columns)))
         .withColumn("model_version", lit(model_version))
         .withColumn("inference_timestamp", to_timestamp(lit(ts)))
     )
+    {% endif %}
     output_df.display()
     # Model predictions are written to the Delta table provided as input.
     # Delta is the default format in Databricks Runtime 8.0 and above.
-    output_df.write.format("delta").mode("overwrite").saveAsTable(output_table_name)
+    output_df.write.format("delta").mode("overwrite").saveAsTable(output_table_name)
```
Comment (arpitjasa-db) [inline {{cookiecutter.project_name}}/steps/predict.py:38]: newline

File: {{cookiecutter.project_name}}/steps/predict.py [hunk 3]
```diff
@@ -10,15 +10,29 @@ def predict_batch(
     writing results to the table with name output_table_name
     """
     table = spark_session.table(input_table_name)
-    predict = mlflow.pyfunc.spark_udf(
```
Comment (arpitjasa-db) [inline {{cookiecutter.project_name}}/steps/predict.py:13]: Shouldn't this be a part of the `else`?

**Pull request merged**